### PR TITLE
Buildsystem: platform CMakeListst.txt, utils CMakeListst.txt. List header files. Auto endianess.

### PR DESCRIPTION
**Pull request by davidgraeff**
Fixes issues #126 #127.

* Platform and examples/tools have their own "CMakeLists.txt" each: Analogous to the core library. Avoids listing source files multiple times: Refactorings are easier.
* Use includeable cmake files for core/ platforms/ and examples/tools instead of CMakeLists.txt. There are no targets defined like it is typically done in a CMakeLists.txt, but source files listed only.

* List header files, too.
* Automatically determine endianess.
* Use CTest feature of cmake in tests/CMakeLists.txt by using enable_testing() and add_test(): Target test is available.

### CODE REVIEW

File: core/wakaama.cmake [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    ${WAKAAMA_SOURCES_DIR}/management.c
+    ${WAKAAMA_SOURCES_DIR}/observe.c
+    ${WAKAAMA_SOURCES_DIR}/json.c
+    ${WAKAAMA_SOURCES_DIR}/discover.c
+    ${CORE_HEADERS}
+    ${EXT_SOURCES})
+
+# This will not work for multi project cmake generators like the Visual Studio Generator
+if(CMAKE_BUILD_TYPE MATCHES Debug)
+   set(WAKAAMA_DEFINITIONS -DLWM2M_WITH_LOGS PARENT_SCOPE)
```
Comment (dnav) [inline core/wakaama.cmake]: This does not work when building in Debug mode.
```cmake -DCMAKE_BUILD_TYPE=Debug ../../examples/lightclient/```

**Comment by dnav** [comment]
Hi David,

sorry but cmake fails for me when building in Debug mode.

Regards

**Comment by davidgraeff** [comment]
PARENT_SCOPE is wrong there indeed, I have extended the pull request.
Is there a reason for a ``LWM2M_WITH_LOGS`` and a ``WITH_LOGS`` flag?

**Comment by dnav** [comment]
It's fine with me.

```LWM2M_WITH_LOGS``` is a flag for the core. For historical reasons the examples are using the ```WITH_LOGS``` flag.

**Pull request merged**